### PR TITLE
added prometheus stats to gloo pods

### PR DESCRIPTION
**Pull request by EItanya**

### CODE REVIEW

**Comment by EItanya** [comment]
@ilackarms where are those docs?

File: install/helm/gloo/templates/3-gloo-deployment.yaml [hunk 1]
```diff
@@ -15,9 +15,11 @@ spec:
     metadata:
       labels:
         gloo: gloo
-      {{- if .Values.gloo.deployment.annotations }}
+      {{- if .Values.gloo.deployment.stats }}
```
Comment (ilackarms) [inline install/helm/gloo/templates/3-gloo-deployment.yaml]: are we missing an `{{ end }}` here?

File: install/helm/gloo/templates/3-gloo-deployment.yaml [hunk 2]
```diff
@@ -15,9 +15,11 @@ spec:
...
       annotations:
-{{ toYaml .Values.gloo.deployment.annotations | indent 8}}{{- end}}
+        prometheus.io/path: /metrics
+        prometheus.io/port: "9091"
+        prometheus.io/scrape: "true"{{- end}}
```
Comment (ilackarms) [inline install/helm/gloo/templates/3-gloo-deployment.yaml]: nit: newline

File: install/helm/gloo/templates/3-gloo-deployment.yaml [hunk 3]
```diff
@@ -28,6 +30,10 @@ spec:
           name: grpc
           protocol: TCP
         args: ["--namespace", "{{.Release.Namespace}}" ]
+        {{- if .Values.gloo.deployment.stats }}
+        env:
+          - name: START_STATS_SERVER
+            value: "true"{{- end}}
```
Comment (ilackarms) [inline install/helm/gloo/templates/3-gloo-deployment.yaml]: nit: newline for {{end}}

File: install/helm/gloo/templates/5-discovery-deployment.yaml [hunk 4]
```diff
@@ -15,10 +15,19 @@ spec:
     metadata:
       labels:
         gloo: discovery
+      {{- if .Values.discovery.deployment.stats }}
+      annotations:
+        prometheus.io/path: /metrics
+        prometheus.io/port: "9091"
+        prometheus.io/scrape: "true"{{- end}}
```
Comment (ilackarms) [inline install/helm/gloo/templates/5-discovery-deployment.yaml]: nit: newline

File: install/helm/gloo/templates/5-discovery-deployment.yaml [hunk 5]
```diff
@@ -15,10 +15,19 @@ spec:
...
     spec:
       containers:
       - image: "{{ .Values.discovery.deployment.image.repository }}:{{ .Values.discovery.deployment.image.tag }}"
         imagePullPolicy: {{ .Values.discovery.deployment.image.pullPolicy }}
         name: discovery
         args: ["--namespace={{.Release.Namespace}}" ]
+        {{- if .Values.discovery.deployment.stats }}
+        env:
+          - name: START_STATS_SERVER
+            value: "true"{{- end}}
```
Comment (ilackarms) [inline install/helm/gloo/templates/5-discovery-deployment.yaml]: nit: newline

**Pull request merged**